### PR TITLE
drivers: ieee802154: Udpate nRF IEEE 802.15.4 radio driver

### PR DESCRIPTION
**Pull request by jciupis**
This commit updates the revision of hal_nordic component
and introduces necessary changes to the IEEE 802.15.4 driver
to match latest nRF IEEE 802.15.4 radio driver API.

Signed-off-by: Jedrzej Ciupis <jedrzej.ciupis@nordicsemi.no>

### CODE REVIEW

File: drivers/ieee802154/ieee802154_nrf5.c [hunk 1]
```diff
@@ -703,8 +703,11 @@ void nrf_802154_received_timestamp_raw(uint8_t *data, int8_t power, uint8_t lqi,
 	__ASSERT(false, "Not enough rx frames allocated for 15.4 driver");
 }
 
-void nrf_802154_receive_failed(nrf_802154_rx_error_t error)
+void nrf_802154_receive_failed(nrf_802154_rx_error_t error, uint32_t id)
 {
+	/* Only immediate reception windows are supported. */
+	__ASSERT_NO_MSG(id == NRF_802154_RESERVED_IMM_RX_WINDOW_ID);
```
Comment (e-rk) [inline drivers/ieee802154/ieee802154_nrf5.c:709]: I don't think assert is needed. I think it is enough to just ignore the id argument.

**Review by e-rk — APPROVED** [review]
I have one comment, but approving regardless.

**Review by rlubos — APPROVED** [review]

**Comment by carlescufi** [comment]
@jciupis please update with a SHA

File: drivers/ieee802154/ieee802154_nrf5.c (hunk 1 above)
Comment (jciupis) [reply drivers/ieee802154/ieee802154_nrf5.c]: Fixed.

**Comment by jciupis** [comment]
@carlescufi done.

**Review by e-rk — APPROVED** [review]
Perfect :+1:

**Pull request merged**